main.go

```diff
@@ -306,14 +306,34 @@ func getCommands() ([]cli.Command, error) {
 	if err != nil {
 		return nil, err
 	}
+
 	allCommands := append(slices.Clone(cliNameSpaces), securityCmds...)
-	allCommands = append(allCommands, artifactoryCmds...)
+	allCommands = appendNewCommands(allCommands, artifactoryCmds)
 	allCommands = append(allCommands, platformServicesCmds...)
 	allCommands = append(allCommands, utils.GetPlugins()...)
 	allCommands = append(allCommands, buildtools.GetCommands()...)
 	return append(allCommands, buildtools.GetBuildToolsHelpCommands()...), nil
 }
 
+// AppendNewCommands merges two slices of cli.Command into one, combining subcommands of commands with the same name.
+func appendNewCommands(a, b []cli.Command) []cli.Command {
```

Comment:
Instead of introducing new functionality to merge commands, we can perhaps expose the Artifactory commands from `jfrog-cli-artifactory` without their namespace, and directly append those to the subcommands of the namespace (such as `Subcommands: append(artifactory.GetCommands(), artifactoryCmds...),`)

Reply:
After discussing this f2f, we decided to continue with the current approach and maybe rename the function to `MergeCommands`.